### ISSUE TITLE
'EntryPoint' object is not callable error when using PyBaMM

### ISSUE THREAD

**Issue by minatouma** [spyder-ide/spyder#20243]
### What steps will reproduce the problem?

I'm running pybamm using DFN model but I get this error'EntryPoint' object is not callable, see below script;
import pybamm
experiment = pybamm.Experiment(
    [
         "Discharge at C/20 for 1 hour",
         "Rest for 1 hour"
    ] * 20,
)
model = pybamm.lithium_ion.DFN()
sim = pybamm.Simulation(model, experiment=experiment)
sim.solve()
sim.plot()

## Versions

* Spyder version: 5.3.3  (conda)
* Python version: 3.9.13 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.7
* Operating System: Windows 10

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.31.1;<8.0.0       :  7.31.1 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.1 (OK)
jellyfish >=0.7               :  0.9.0 (OK)
jsonschema >=3.2.0            :  4.16.0 (OK)
keyring >=17.0.0              :  23.4.0 (OK)
nbconvert >=4.0               :  6.4.4 (OK)
numpydoc >=0.6.0              :  1.5.0 (OK)
paramiko >=2.4.0              :  2.8.1 (OK)
parso >=0.7.0;<0.9.0          :  0.8.3 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.9.0 (OK)
pygments >=2.0                :  2.11.2 (OK)
pylint >=2.5.0;<3.0           :  2.14.5 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.5.0;<1.6.0          :  1.5.0 (OK)
pylsp_black >=1.2.0           :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0     :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=1.0.2             :  1.0.3 (OK)
qtconsole >=5.3.2;<5.4.0      :  5.3.2 (OK)
qtpy >=2.1.0                  :  2.2.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  65.5.0 (OK)
sphinx >=0.6.6                :  5.0.2 (OK)
spyder_kernels >=2.3.3;<2.4.0 :  2.3.3 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
zmq >=22.1.0                  :  23.2.0 (OK)

# Optional:
cython >=0.21                 :  0.29.32 (OK)
matplotlib >=3.0.0            :  3.6.2 (OK)
numpy >=1.7                   :  1.23.4 (OK)
pandas >=1.1.1                :  1.4.4 (OK)
scipy >=0.17.0                :  1.9.3 (OK)
sympy >=0.7.3                 :  1.11.1 (OK)
```

**Comment by dalthviz** [comment]
Hi @minatouma seems like this is an error with `pybamm` https://github.com/pybamm-team/PyBaMM/issues/2534 ? Just in case, could you try to run your code using IPython and check if the same issue appears? For that open an Anaconda prompt, activate the env Spyder is installed and then run from there `ipython`. Let us know!

**Comment by tinosulzer** [comment]
@dalthviz this comment shows the problem: https://github.com/pybamm-team/PyBaMM/issues/2534#issuecomment-1363815564

The entry points are created here: https://github.com/pybamm-team/PyBaMM/blob/c4baa51081f7ba38eea6ccb21a9e9fe89b71c934/pybamm/parameters/parameter_sets.py#L39-L42
```python3
for entry_point in importlib_metadata.entry_points(
            group="pybamm_parameter_sets"
        ):
            self.__all_parameter_sets[entry_point.name] = entry_point
```
which in most environments returns an `imporlib_metdata.EntryPoint` object that we then load here:
https://github.com/pybamm-team/PyBaMM/blob/c4baa51081f7ba38eea6ccb21a9e9fe89b71c934/pybamm/parameters/parameter_sets.py#L60-L61
```python3
if isinstance(ps, importlib_metadata.EntryPoint):
            ps = self.__all_parameter_sets[key] = ps.load()
```
But in spyder the object returned in the first step is a `setuptools._vendor.importlib_metadata.EntryPoint`, so it doesn't get caught by the second if statement

How would you suggest we resolve this? Add a second `isintance()` check for the spyder type or something different?

**Comment by dalthviz** [comment]
Hi @tinosulzer thank you for the explanation! Adding another `isinstance` sounds good to me but just in case, what do you think @ccordoba12 ?

**Comment by ccordoba12** [comment]
Yeah, sounds good to me too. And thanks @tinosulzer for chiming in and be interested in fixing this issue on behalf of Spyder users.

**Comment by minatouma** [comment]
Thanks so much for your support. However, I'm still getting the same error with Spyder, may be I'm not sure how to incorporate the code with pybamm script above to fix the issue

**Comment by ccordoba12** [comment]
@minatouma, this needs to be fixed in pybamm and released in a new version of it. Until that happens, I'm afraid there's nothing you can do about it.

**Comment by tinosulzer** [comment]
Working on the fix @minatouma but until then just use `v22.10` or earlier

**Comment by dalthviz** [comment]
Closing since the fix was released some time ago. See https://github.com/pybamm-team/PyBaMM/releases/tag/v22.12
